web-client/public/js/graph.js

```diff
@@ -170,44 +207,65 @@ var drawGraph = function (nodes, links, positiveWeights, negativeWeights, sheetT
   });
 
   d3.selectAll("input[name=viewport]").on("change", function () {
-    let fixed = $(this).prop("checked");
-    if (!adaptive && !fixed) {
+    var fixed = $(this).prop("checked");
+    if (!fixed) {
+      if (!scrolling) {
+        $container.addClass("cursorGrab");
+        scrolling = true;
+      }
       adaptive = true;
-      MAX_SCALE = 10;
+      maximumScale = ADAPTIVE_MAX_SCALE;
+      zoom.scaleExtent([minimumScale, maximumScale])
+      d3.select(".zoomSlider").attr("max", maximumScale);
+
       d3.select("rect").attr("stroke", "none");
-      zoom.scaleExtent([MIN_SCALE, MAX_SCALE])
-      d3.select(".zoomSlider").attr("max", MAX_SCALE);
-    } else if (adaptive && fixed) {
-      var newWidth = $(".grnsight-container").css("width");
-      var newHeight = $(".grnsight-container").css("height");
+    } else if (fixed) {
+      adaptive = false;
+      maximumScale = 1;
+      zoom.scaleExtent([minimumScale, maximumScale]);
+      if (zoom.scale() >= 1) {
+        zoom.scale(1);
+        scrolling = false;
+        if ($container.hasClass("cursorGrabbing")) $container.removeClass("cursorGrabbing");
+        if ($container.hasClass("cursorGrab")) $container.removeClass("cursorGrab");
+      }
+      d3.select(".zoomSlider").attr("max", maximumScale);
+
+      var newWidth = $container.css("width");
+      var newHeight = $container.css("height");
       width = newWidth.substring(0, newWidth.length - 2) - BORDER_OFFSET;
       height = newHeight.substring(0, newHeight.length - 2) - BORDER_OFFSET;
-
-      adaptive = false;
-      MAX_SCALE = 1;
       d3.select("rect").attr("stroke", "#9A9A9A").attr("width", width).attr("height", height);
       $(".boundingBox").attr("width", width).attr("height", height);
-      zoom.scaleExtent([MIN_SCALE, MAX_SCALE]);
-      if (zoom.scale() > 1) {
-          scale(1);
-      }
-      d3.select(".zoomSlider").attr("max", MAX_SCALE);
 
-      force.size([width, height]).resume();
+      center();
     }
+    force.size([width, height]).resume();
   });
 
   $(window).on("resize", function () {
-      if ( $(".grnsight-container").hasClass("containerFit")) {
+      if ($container.hasClass("containerFit")) {
           $(".boundBoxSize").trigger("click");
       }
   });
 
+  $container.on("mousedown", function () {
+     if (scrolling) {
+       if ($container.hasClass("cursorGrab")) $container.removeClass("cursorGrab");
+       if (!$container.hasClass("cursorGrabbing")) $container.addClass("cursorGrabbing");
+     }
+  }).on("mouseup", function () {
+    if (scrolling) {
+      if ($container.hasClass("cursorGrabbing")) $container.removeClass("cursorGrabbing");
+      if (!$container.hasClass("cursorGrab")) $container.addClass("cursorGrab");
```

Comment:
More line break needs; but actually, having now seen this multiple times, it sounds like you need two helper functions:

    removeClassIfPresent($element, className)
    addClassIfAbsent($element, className)

.....in fact, I think that jQuery is actually pretty robust here. You might not need the condition check if removing or adding classes. Removing a non-existent class should have no effect, and neither should adding an already-existent class.

I suggest that you do a quick experiment to suss out the behavior of `removeClass` and `addClass` under these conditions, then adjust the code accordingly.